bahmni/src/main/java/org/avni_integration_service/bahmni/service/SubjectService.java

```diff
@@ -142,10 +138,33 @@ public Subject createSubjectFromPatient(OpenMRSPatient patient, Constants consta
             subject.addObservation(metaData.avniIdentifierConcept(), identifier.get().getIdentifier().replace(prefix, ""));
         }
 
+        List<OpenMRSPatientIdentifier> identifiers = patient.getIdentifiers();
+        Optional<OpenMRSPatientIdentifier> rchIdentifier = patient.getIdentifiers().stream()
+                .filter(id -> id.getIdentifierType().getUuid().equals("3766473c-0c29-11ee-be56-0242ac120002"))
+                .findFirst();
+        if (rchIdentifier.isPresent()) {
+            subject.addObservation("RCH ID", identifiers.get(0).getIdentifier());
```

Comment:
Using `identifiers.get(0).getIdentifier()` is not a good approach because if one of the identifier is not filled, it will lead to exceptions. You can use `rchIdentifier.get().getIdentifier()`